src/librustc_mir/transform/generator.rs

```diff
@@ -1314,3 +1340,130 @@ impl<'tcx> MirPass<'tcx> for StateTransform {
         create_generator_resume_function(tcx, transform, source, body, can_return);
     }
 }
+
+/// Looks for any assignments between locals (e.g., `_4 = _5`) that will both be converted to fields
+/// in the generator state machine but whose storage does not conflict.
```

Comment:
At first I was confused by the wording here. To me the storage _does_ conflict, we just aren't aware of it. Maybe:

```suggestion
/// in the generator state machine but whose storage is not marked as conflicting.
```